### DISCUSSION TITLE
Kodi use Just Player for Dolby Vision playback

### DISCUSSION THREAD

**Discussion by KyleSanderson** [moneytoo/Player#310] [Show and tell]
as Kodi in the year of our lord 2022 still does not support Dolby Vision mode-switching in-tree (or even on stable forks) - here's what I've picked up over the last month or so of trying **every single player** under the sun (Jellyfin, Emby, Plex, Nova, BSPlayer, VLC, etc). 

Kodi has full support for nginx directory listings over both http and tls. Historically this was only supported by Apache, however there's no documentation for it but nginx and code for reading directory listings from a few other webservers has since been merged long ago. This is the best way to actually use Kodi in a performant manner on Android. While HTTP is the best performance option available, if you still need to use ssh, tunnel the traffic using autossh.

Now that that's settled - it's actually quite easy to get this done.

Create the following `playercorefactory.xml` file using the editor of your choice.

```
<playercorefactory>
	<players>
		<player name="JustPlayer" type="ExternalPlayer" audio="false" video="true">
			<filename>com.brouken.player</filename>
			<hidexbmc>true</hidexbmc>
			<playcountminimumtime>120</playcountminimumtime>
		</player>
	</players>
    <rules action="prepend">
        <rule video="true" player="dvdplayer">
            <rule filename=".*DV.*|.*DoVi.*|.*Dolby.Vision.*" player="JustPlayer"/>
        </rule>
    </rules>
</playercorefactory>
```

Now we're going to open Kodi on your device. Make sure you're at the main menu after allowing all kinds of permissions required for startup. This step is important, because it's going to create the directory structure on your filesystem.

Next, connect to your device using adb, use adb shell and browse to `/sdcard/Android/data` to find your package name (this step will age with time - you're finding the install path of kodi). Once you have the package name, exit out of the shell.

Now that we have the package name and path, open another terminal window and lets push the file to our device.
`adb push playercorefactory.xml /sdcard/Android/data/DISTRO_HERE/files/.kodi/userdata`
* Note the *DISTRO_HERE* should be replaced with your kodi distribution, mine is `net.kodinerds.maven.kodi.firetv`.

Finally, in the Kodi GUI, go down to the power button, and exit the application.

That's it - now you're using JustPlayer for Dolby Vision files. There are additional nobs and dials, as well as other players, and filenames that can be configured as Kodi continues to both lead and lag behind in this area.

Documentation on playercorefactory can be found here: https://kodi.wiki/view/External_players

**Comment by Kiwimannen** [comment]
Hi, im trying to get this to work with version 21.2 of Kodi  Android on a Lenovo P11 Pro, with JustPlayer 0.192  (since Kodi cant play Dolby Vision on the P11 Pro but JustPlayer can)

But JustPlayer never starts - is there something that needs to be updated ?
